pennylane/capture/base_interpreter.py

```diff
@@ -0,0 +1,415 @@
+        )
+        return primitive.bind(*invals, **params)
+
+    def eval(self, jaxpr: "jax.core.Jaxpr", consts: list, *args) -> list:
+        """Evaluate a jaxpr.
+
+        Args:
+            jaxpr (jax.core.Jaxpr): the jaxpr to evaluate
+            consts (list[TensorLike]): the constant variables for the jaxpr
+            *args (tuple[TensorLike]): The arguments for the jaxpr.
+
+        Returns:
+            list[TensorLike]: the results of the execution.
+
+        """
+        self._env = {}
+        self._op_math_cache = {}
+        self.setup()
+
+        for arg, invar in zip(args, jaxpr.invars):
+            self._env[invar] = arg
+        for const, constvar in zip(consts, jaxpr.constvars):
+            self._env[constvar] = const
+
+        for eqn in jaxpr.eqns:
+
+            custom_handler = self._primitive_registrations.get(eqn.primitive, None)
+            if custom_handler:
+                invals = [self.read(invar) for invar in eqn.invars]
+                outvals = custom_handler(self, *invals, **eqn.params)
+            elif isinstance(eqn.outvars[0].aval, AbstractOperator):
+                outvals = self.interpret_operation_eqn(eqn)
+            elif isinstance(eqn.outvars[0].aval, AbstractMeasurement):
+                invals = [self.read(invar) for invar in eqn.invars]
+                outvals = self.interpret_measurement_eqn(eqn.primitive, *invals, **eqn.params)
+            else:
+                invals = [self.read(invar) for invar in eqn.invars]
+                outvals = eqn.primitive.bind(*invals, **eqn.params)
+
+            if not eqn.primitive.multiple_results:
+                outvals = [outvals]
+            for outvar, outval in zip(eqn.outvars, outvals):
+                self._env[outvar] = outval
+
+        # Read the final result of the Jaxpr from the environment
+        outvals = []
+        for var in jaxpr.outvars:
+            if var in self._op_math_cache:
+                outvals.append(self.interpret_operation(self._op_math_cache[var]))
+            else:
+                outvals.append(self.read(var))
+        self.cleanup()
+        self._op_math_cache = {}
+        self._env = {}
+        return outvals
+
+    def __call__(self, f: Callable) -> Callable:
+
+        flat_f = FlatFn(f)
+
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            with qml.QueuingManager.stop_recording():
+                jaxpr = jax.make_jaxpr(partial(flat_f, **kwargs))(*args)
+            results = self.eval(jaxpr.jaxpr, jaxpr.consts, *args)
+            assert flat_f.out_tree
+            return jax.tree_util.tree_unflatten(flat_f.out_tree, results)
+
+        return wrapper
+
+
+@PlxprInterpreter.register_primitive(adjoint_transform_prim)
+def handle_adjoint_transform(self, *invals, jaxpr, lazy, n_consts):
+    """Interpret an adjoint transform primitive."""
+    consts = invals[:n_consts]
+    args = invals[n_consts:]
+
+    jaxpr = jaxpr_to_jaxpr(type(self)(), jaxpr, consts, *args)
```

Comment:
So interpreting an `adjoint` instantiates a new interpreter? That's funky :sweat_smile: